google/web-stories-wp
# Media: Skip video optimization based on criteria

## Context



We want to avoid needlessly running video optimization on videos that are already in very good shape, for example when the creator has manually optimized videos before upload.

The aim is to improve overall user experience and performance during uploads.

## Summary



## Relevant Technical Choices



Uses [mediainfo.js](https://github.com/buzz/mediainfo.js) to get detailed information about codecs, file size, frame rate, etc.

For the start, small videos (< 5MB), with either MP4 or WEBM type and max dimensions of 720x1280, will be automatically marked as optimized.

## To-do



## User-facing changes



No visual changes per se, but when uploading smaller videos fitting the criteria they will not be optimized by default anymore.

## Testing Instructions






This PR can be tested by following these steps:

1. Enable video optimization
2. Upload small videos (< 5MB), with either MP4 or WEBM type and max dimensions of 720x1280
2. Notice that video is not automatically optimized, but uploaded straight away
3. Notice that the checklist does not mark the video as un-optimized
4. Verify that larger videos are still going through optimization
5. Disable video optimization
6. Verify that uploads still work. Smaller videos are marged as optimized, while larger videos and videos of other types (e.g. MOV) are rejected

## Reviews

### Does this PR have a security-related impact?



No

### Does this PR change what data or activity we track or use?



No

### Does this PR have a legal-related impact?



No

## Checklist




---



Fixes #11405
